nfadeeva/assignment4/count_islands.py

```diff
@@ -0,0 +1,36 @@
+from typing import List
+
+
+def neighbors(input_array, num_rows, num_cols, x, y):
+    for dx, dy in [(-1, 0), (0, -1), (0, 1), (1, 0)]:
+            xn, yn = x + dx, y + dy
+            if (0 <= xn < num_rows and 0 <= yn < num_cols and
+                    input_array[xn][yn]):
+                yield xn, yn
+
+
+def dfs(num_rows, num_cols, input_array, x, y):
+
+    # mark the current tile as visited
+    input_array[x][y] = False
+    # start dfs from every neighbour of the current tile
+    for next_x, next_y in neighbors(input_array, num_rows, num_cols, x, y):
+        dfs(num_rows, num_cols, input_array, next_x, next_y)
+
+
+def count_islands(num_rows, num_cols, input_array: List[bool]):
```

Comment:
Is input_array a list of bool or a list of lists of bool?
Try running your code through mypy:
```
$ mypy count_islands.py
```